src/api/api.ts

```diff
@@ -252,18 +252,31 @@ async function getIncompleteChecks(names?: string[]): Promise<string[]> {
 }
 
 export async function waitForChecks(names?: string[]) {
```

Comment:
Two improvements here:

 - I am now returning a `cancel` function that should be called when unmounting the component, to avoid further requests for checks when the user is no longer in the page.
 - I am catching errors, which avoids this to hang forever if `getChecks` throws an error.